src/libs/migrations/Participants.ts

```diff
@@ -0,0 +1,83 @@
+import Onyx from 'react-native-onyx';
+import type {NullishDeep, OnyxCollection} from 'react-native-onyx';
+import Log from '@libs/Log';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Report} from '@src/types/onyx';
+import type {Participants} from '@src/types/onyx/Report';
+
+type ReportKey = `${typeof ONYXKEYS.COLLECTION.REPORT}${string}`;
+type OldReport = Report & {participantAccountIDs?: number[]; visibleChatMemberAccountIDs?: number[]};
+type OldReportCollection = Record<ReportKey, NullishDeep<OldReport>>;
+
+function getReports(): Promise<OnyxCollection<OldReport>> {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: ONYXKEYS.COLLECTION.REPORT,
+            waitForCollectionCallback: true,
+            callback: (reports) => {
+                Onyx.disconnect(connectionID);
+                return resolve(reports);
+            },
+        });
+    });
+}
+
+function getCurrentUserAccountID(): Promise<number | undefined> {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: ONYXKEYS.SESSION,
+            callback: (session) => {
+                Onyx.disconnect(connectionID);
+                return resolve(session?.accountID);
+            },
+        });
+    });
+}
+
+export default function (): Promise<void> {
+    return Promise.all([getCurrentUserAccountID(), getReports()]).then(([currentUserAccountID, reports]) => {
+        if (!reports) {
+            Log.info('[Migrate Onyx] Skipped Participants migration because there are no reports');
+            return;
+        }
+
+        const collection = Object.entries(reports).reduce<OldReportCollection>((reportsCollection, [onyxKey, report]) => {
+            // If we have participantAccountIDs then this report is eligible for migration
+            if (report?.participantAccountIDs) {
```

Comment:
The condition here will be true if  `participantAccountIDs` is empty array `[]` (no need for migration in this case), and the migration in this case will remove valid accountIDs in participants in Line 65 `participants[accountID] = null`, and this happened after login with new accounts, and we handled it here https://github.com/Expensify/App/pull/45369

Reply:
Why is `participantAccountIDs` an empty array in the first place? That was the source of truth before migrating and after migrating it should be `null`.

Reply:
> Why is participantAccountIDs an empty array in the first place?

I think backend should not return `participantAccountIDs` or return it with `null` value for this case (new logged user), because the returned data is already migrated. But may be the backend doing that to prevent old version from crashes and unexpected errors

Reply:
`participantAccountIDs ` shouldn't be returned by the backend indeed. cc @marcaaron Maybe you can double check this one?

Reply:
I agree this will be better if it is not affected on old versions, but any way we add extra check in FE for empty array here https://github.com/Expensify/App/pull/45369

Reply:
We internally paused the project related to this so I think doing a frontend solution is fine until I am free to clean it up the whole way. Took a quick look and it's possible for this to be an empty array - but unlikely/impossible for it to have accountID values in it and we should still use the `report.participants` object in all cases. Thanks!